csharp/src/Apache.Arrow.Adbc/AdbcConnection11.cs

```diff
@@ -0,0 +1,390 @@
+#if NET5_0_OR_GREATER
+        , IAsyncDisposable
+#endif
+    {
+        ~AdbcConnection11() => Dispose(false);
+
+        /// <summary>
+        /// Attempts to cancel an in-progress operation on a connection.
+        /// </summary>
+        /// <remarks>
+        /// This can be called during a method like GetObjects or while consuming an ArrowArrayStream
+        /// returned from such. Calling this function should make the other function throw a cancellation exception.
+        ///
+        /// This must always be thread-safe.
+        /// </remarks>
+        public virtual void Cancel()
+        {
+            throw AdbcException.NotImplemented("Connection does not support cancellation");
+        }
+
+        /// <summary>
+        /// Starts a new transaction with the given isolation level
+        /// </summary>
+        /// <param name="isolationLevel">The isolation level for the new transaction.</param>
+        public virtual void BeginTransaction(IsolationLevel? isolationLevel = default)
+        {
+            Task.Run(() => BeginTransactionAsync(isolationLevel)).GetAwaiter().GetResult();
+        }
+
+        public virtual Task BeginTransactionAsync(IsolationLevel? isolationLevel = default, CancellationToken cancellationToken = default)
+        {
+            throw AdbcException.NotImplemented("Connection does not support transactions");
+        }
+
+        /// <summary>
+        /// Create a new statement to bulk insert into a table.
+        /// </summary>
+        /// <param name="targetCatalog">The catalog name, or null to use the current catalog</param>
+        /// <param name="targetDbSchema">The schema name, or null to use the current schema</param>
+        /// <param name="targetTableName">The table name</param>
+        /// <param name="mode">The ingest mode</param>
+        /// <param name="isTemporary">True for a temporary table. Catalog and Schema must be null when true.</param>
+        public virtual AdbcStatement BulkIngest(string? targetCatalog, string? targetDbSchema, string targetTableName, BulkIngestMode mode, bool isTemporary)
+        {
+            throw AdbcException.NotImplemented("Connection does not support BulkIngest");
+        }
+
+        /// <summary>
+        /// Commit the pending transaction.
+        /// </summary>
+        public virtual void Commit()
+        {
+            Task.Run(() => CommitAsync()).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Commit the pending transaction.
+        /// </summary>
+        public virtual Task CommitAsync(CancellationToken cancellationToken = default)
+        {
+            throw AdbcException.NotImplemented("Connection does not support transactions");
+        }
+
+        /// <summary>
+        /// Create a new statement that can be executed.
+        /// </summary>
+        public abstract AdbcStatement CreateStatement(IReadOnlyDictionary<string, object>? options = default);
```

Comment:
this isn't an expensive operation, but does it need an Async equivalent for consistency?

Reply:
In my mind, part of the goal of the API is to signal to implementers what they should be doing. I see a statement as a kind of unit of internal bookkeeping which doesn't use database resources until you ask it to do something else. By not having an async function for creating a statement, this intent is communicated fairly clearly to implementers. This is also consistent with ADO.NET's DbCommand, which doesn't have an async construction function (though there were likely other considerations there that we're not faced with).

If there's a demonstrated need to do something differently, we can later add an async method that returns a ValueTask and that would be consistent with everything else we have.

That said, I'm mostly neutral on adding it now so if you think it's better to have it then I'll change it.